### DISCUSSION TITLE
Wrapping structs with common type and methods

### DISCUSSION THREAD

**Discussion by xandox** [PyO3/pyo3#2986] [Questions]
Hello. 
I have a question.
I have one crate with something like:
```rust
pub struct A {
// ...
}

pub struct B {
   a: A
}

pub struct C {
   a: A
}

impl A {
// ...
}

impl B {
// ...
}

impl C {
// ...
}

// and impl Deref, DerefMut for B and C with Target=A
```

In other crate I would wrap it in python classes.
And question is how to make this subclassing? 
In subclassing examples structs do not share a data. 
Right now I duplicate methods for A in pyclass wrapers for B and C. But it's look ugly and bad at all.

**Comment by davidhewitt** [comment]
This is related to the discussion in https://github.com/PyO3/pyo3/issues/1637#issuecomment-848662144 .

In all honesty we should think more about this case, because some users will want to do it (and it's a natural analogy of Python / C's inheritance mechanism). But to do so requires quite some work in the internals of PyO3. Would be happy to chat more about this!
